cmd/rb-ldap/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/redbrick/rbldap/internal/pkg/rbldap"
+	"github.com/urfave/cli"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "rb-ldap"
+	app.Usage = "Command line interface for Redbrick Ldap"
+	app.ArgsUsage = ""
+	app.HideVersion = true
+	app.EnableBashCompletion = true
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "user, u",
+			Value: "cn=root,ou=ldap,o=redbrick",
+			Usage: "ldap user, used for authentication",
+		},
+
+		cli.StringFlag{
+			Name:  "dcu-user",
+			Value: "CN=rblookup,OU=Service Accounts,DC=ad,DC=dcu,DC=ie",
+			Usage: "Active Directory user for DCU, used for authentication",
+		},
+
+		cli.StringFlag{
+			Name:  "host",
+			Value: "ldap.internal",
+			Usage: "ldap host to query",
+		},
+
+		cli.StringFlag{
+			Name:  "dcu-host",
+			Value: "ad.dcu.ie",
+			Usage: "DCU Active Directory host to query",
+		},
+
+		cli.IntFlag{
+			Name:  "port, p",
+			Value: 389,
+			Usage: "Port for ldap host",
+		},
+
+		cli.IntFlag{
+			Name:  "dcu-port",
+			Value: 389,
+			Usage: "Port for DCU Active Directory host",
+		},
+
+		cli.StringFlag{
+			Name:     "password",
+			Usage:    "password for the ldap server",
+			FilePath: "/etc/ldap.secret",
+		},
+
+		cli.StringFlag{
+			Name:     "dcu-password",
+			Usage:    "password for the DCU ldap server",
+			FilePath: "/etc/dcu_ldap.secret",
+		},
```

Comment:
Even another idea is to combine all those flags into `user:pass@host:port`. I feel like `url.URL` would help with parsing. It makes the flags less awkward to use.

Reply:
for the squashed syntax again see above.
explain url.URL

Reply:
See https://play.golang.org/p/KAornm0cci1

You let the flag be parsed `url.Parse`. I understand the only problematic bit would be the password, which is understandably another CLI flag loading the password from file.

Reply:
id **MUCH** rather keep using the filepath for a password file rather then a password

Reply:
that makes changing the variables seem pretty annoying

Reply:
Even if passwords are loaded from file, once we have sane defaults, how often would you actually need to specify  it yourself? 

Keeping password separate we go down from 8 flags total for RB and DCU LDAP, to just 4 (user@rb:port, user@dcu:port, rb-password, dcu-password). Honestly I have never encountered decent application which forces you to change 4 different flags to log in (and I am not counting mysql as a good example).

But this is just my opinion, I think settling this rests on cmt hands :)

Reply:
admins, not cmt 
you aren't likely to be using any flags though the default will work out of the box the flags are only for if something changes

Reply:
Admin here, hi. Path to file is how this is happening, non-negotiable, I'd really prefer any user couldn't run a `ps` and see our pass being passed as an arg.